test/specs/json_ld/tests/01.reference_google_dataset.js

```diff
@@ -0,0 +1,84 @@
+
+exports.execute = function (options) {
+    var module = options.includes.ermRest;
+    
+    describe('For determining google_metadata annotation, ', function () {
+        var catalog_id = process.env.DEFAULT_CATALOG,
+            schemaName = "google_metadata_schema",
+            tableName1 = "google_metadata_w_handlebars";
+
+        var handlebarsGoogleMetadataUri = options.url + "/catalog/" + catalog_id + "/entity/" + schemaName + ":" + tableName1 + "/@sort(id)";
+
+        var chaiseURL = "https://dev.isrd.isi.edu/chaise";
+        var recordURL = chaiseURL + "/record";
+        var record2URL = chaiseURL + "/record-two";
+        var viewerURL = chaiseURL + "/viewer";
+        var searchURL = chaiseURL + "/search";
+        var recordsetURL = chaiseURL + "/recordset";
+
+        var appLinkFn = function (tag, location) {
+            var url;
+            switch (tag) {
+                case "tag:isrd.isi.edu,2016:chaise:record":
+                    url = recordURL;
+                    break;
+                case "tag:isrd.isi.edu,2016:chaise:record-two":
+                    url = record2URL;
+                    break;
+                case "tag:isrd.isi.edu,2016:chaise:viewer":
+                    url = viewerURL;
+                    break;
+                case "tag:isrd.isi.edu,2016:chaise:search":
+                    url = searchURL;
+                    break;
+                case "tag:isrd.isi.edu,2016:chaise:recordset":
+                    url = recordsetURL;
+                    break;
+                default:
+                    url = recordURL;
+                    break;
+            }
+
+            url = url + "/" + location.path;
+
+            return url;
+        };
+
+        beforeAll(function () {
+            options.ermRest.appLinkFn(appLinkFn);
+        });
+
+        describe("with handlebar templating", function () {
+            var tuples, tuple, reference;
+
+            beforeAll(function (done) {
+                options.ermRest.resolve(handlebarsGoogleMetadataUri, { cid: "test" }).then(function (response) {
+                    reference = response.contextualize.detailed;
+                    return response.read(2);
+                }).then(function (response) {
+                    tuples = response.tuples;
+                    done();
+                }).catch(function (err) {
+                    console.dir(err);
+                    done.fail();
+                });
+            });
+
+            it("The google_metadata with all required attributes is successfully generated from the annotation", function (done) {
+                tuple = tuples[0];
+                var metadata = reference.googleDatasetMetadata.compute(tuple);
+                expect(metadata).toBeObject();
+                expect(JSON.stringify(metadata)).toBeJsonString(JSON.stringify(require("../resources/generated_json_ld.json")));
```

Comment:
I should have mentioned that the value of RID is determined by ermrest and we have to look at database to get it. To get around this, when we're creating the test data, we're creating a `options.entitites` which can be used to get the RID value of a row. You can see an example of it [here](https://github.com/informatics-isi-edu/ermrestjs/blob/master/test/specs/annotation/tests/03.table_display.js#L69).

Ideally this should be a utility function that you can use everywhere in test cases, but since we don't have that capability now, I suggest adding the same helper function and then using it the same way. 

Assuming you added the same helper function, you could do something like this instead:

```js
var expectedVal = require("../resources/generated_json_ld.json");

// since RID value is generated by database, we have to attach it here
var RIDVal = findRID(tableName1, "2003");
expectedVal.url = "https://dev.isrd.isi.edu/chaise/record/google_metadata_schema:google_metadata_w_handlebars/RID=" + RIDVal;

// TODO compare the objects
```

And about the actual expect statement, does `toBeJsonString` even check the object? I think it only makes sure the given string is json parsable. If you're stringifying both objects, then you could just compare their string values.  Although that would make the debugging very difficult. I think you can use the `toHaveSameItems` custom matcher that is defined in [here](https://github.com/informatics-isi-edu/ermrestjs/blob/master/test/utils/jasmine-matchers.js#L9).

Reply:
Thanks for pointing this out!!